src/Inciter/BoundaryConditions.C

```diff
@@ -0,0 +1,192 @@
+  // that associates a vector of pairs associated to local node IDs. (The pair
+  // is a pair of bool and real value, the former is the fact that the BC is to
+  // be set while the latter is the value if it is to be set). The length of
+  // this NodeBC vector, returning from each system of PDEs equals to the number
+  // of scalar components the given PDE integrates. Here then we contatenate
+  // this map for all PDEs integrated. If there are multiple BCs set at a mesh
+  // node (dirbc::key), either because (1) in the same PDE system the user
+  // prescribed BCs on side sets that share nodes or (2) because more than a
+  // single PDE system assigns BCs to a given node (on different variables), the
+  // NodeBC vector must be correctly stored. "Correctly" here means that the
+  // size of the NodeBC vectors must all be the same and qual to the sum of all
+  // scalar components integrated by all PDE systems integrated. Example:
+  // single-phase compressible flow (density, momentum, energy = 5) +
+  // transported scalars of 10 variables -> NodeBC vector length = 15. Note that
+  // in case (1) above a new node encountered must "overwrite" the already
+  // existing space for the NodeBC vector. "Overwrite" here means that it should
+  // keep the existing BCs and add the new ones yielding the union the two
+  // prescription for BCs but in the same space that already exist in the NodeBC
+  // vector. In case (2), however, the NodeBC pairs must go to the location in
+  // the vector assigned to the given PDE system, i.e., using the above example
+  // BCs for the 10 (or less) scalars should go in the positions starting at 5,
+  // leaving the first 5 false, indicating no BCs for the flow variables.
+  //
+  // TODO: Note that the logic described above is only partially implemented at
+  // this point. What works is the correct insertion of multiple BCs for nodes
+  // shared among multiple side sets, e.g., corners, originating from the same
+  // PDE system. What is not yet implemented is the case when there are no BCs
+  // set for flow variables but there are BCs for transport, the else branch
+  // below will incorrectly NOT skip the space for the flow variables. In other
+  // words, this only works for a single PDE system and a sytem of systems. This
+  // machinery is only tested with a single system of PDEs at this point.
+
+  for (const auto& s : sidenodes) {
+    std::size_t c = 0;
+    for (std::size_t eq=0; eq<g_pdes.size(); ++eq) {
+      auto eqbc = g_pdes[eq].dirbc( t, dt, s, coord );
+      for (const auto& n : eqbc) {
+        auto id = n.first;                      // BC node ID
+        const auto& bcs = n.second;             // BCs
+        auto& nodebc = dirbc[ gid[id] ];        // BCs to be set for node
+        if (nodebc.size() > c) {        // node already has BCs from this PDE
+          Assert( nodebc.size() == c+bcs.size(), "Size mismatch" );
+          for (std::size_t i=0; i<bcs.size(); i++)
```

Comment:
Yeah, you are right. I have added the braces for the `for` loop.

A style guide is not a bad idea. It would also be good to enforce this automatically. Can this be done automatically with a customized `clang-format` style in a way that does not result in a huge difference compared to the current formatting?

Reply:
Yeah, you are right. I have added the braces for the for loop.

A style guide is not a bad idea. It would also be good to enforce this automatically. Can this be done automatically with a customized clang-format style in a way that does not result in a huge difference compared to the current formatting?

Reply:
I don't know how to do this, but it may be worth looking into!